src/Rubric/RubSmalltalkEditor.class.st

```diff
@@ -192,15 +192,14 @@ RubSmalltalkEditor >> bestNodeInTextArea [
 	"Find the best node in the editor text area at the current pointer location"
 
 	| node start stop |
-
 	start := self textArea startIndex.
 	stop := self textArea stopIndex.
 
-	[node := RBParser parseMethod: self textArea string onError: [
-		RBParser parseFaultyExpression: self textArea string]
-	] on: Error do: [^nil].
+	[ node := RBParser parseMethod: self textArea string onError: [ :aString | RBParser parseFaultyExpression: aString ] ]
+		on: Error
+		do: [ ^ nil ].
```

Comment:
this is a bad use of the parser API (that is also bad, and unintuitive).

The correct code might be to replace the whole thing by a simpler

```st
RBParser parseFaultyExpression: self textArea string.
```

without `onError:` not `on:do:`

Reply:
I'll do that thanks